src/gui/src/staDescriptors.cpp

```diff
@@ -0,0 +1,498 @@
+
+  while (lib_iter->hasNext()) {
+    sta::LibertyLibrary* library = lib_iter->next();
+    objects.insert(makeSelected(library));
+  }
+
+  return true;
+}
+
+//////////////////////////////////////////////////
+
+LibertyCellDescriptor::LibertyCellDescriptor(odb::dbDatabase* db,
+                                             sta::dbSta* sta)
+    : db_(db), sta_(sta)
+{
+}
+
+std::string LibertyCellDescriptor::getName(std::any object) const
+{
+  return std::any_cast<sta::LibertyCell*>(object)->name();
+}
+
+std::string LibertyCellDescriptor::getTypeName() const
+{
+  return "Liberty cell";
+}
+
+bool LibertyCellDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  return false;
+}
+
+void LibertyCellDescriptor::highlight(std::any object, Painter& painter) const
+{
+}
+
+Descriptor::Properties LibertyCellDescriptor::getProperties(
+    std::any object) const
+{
+  auto cell = std::any_cast<sta::LibertyCell*>(object);
+
+  auto gui = Gui::get();
+  Properties props;
+
+  if (auto area = cell->area()) {
+    props.push_back({"Area", area});
+  }
+  add_if_true(props, "Dont Use", cell->dontUse());
+  props.push_back({"Filename", cell->filename()});
+  add_if_true(props, "Has Sequentials", cell->hasSequentials());
+  add_if_true(props, "Is Always On", cell->alwaysOn());
+  add_if_true(props, "Is Buffer", cell->isBuffer());
+  add_if_true(props, "Is Clock Cell", cell->isClockCell());
+  add_if_true(props, "Is Inverter", cell->isInverter());
+  add_if_true(props, "Is Isolation Cell", cell->isIsolationCell());
+  add_if_true(props, "Is Level Shifter", cell->isLevelShifter());
+  add_if_true(props, "Is Macro", cell->isMacro());
+  add_if_true(props, "Is Memory", cell->isMemory());
+  add_if_true(props, "Is Pad", cell->isPad());
+  props.push_back({"Library", gui->makeSelected(cell->libertyLibrary())});
+
+  std::array<SelectionSet, 8> ports;
+  sta::LibertyCellPortIterator port_iter(cell);
+  while (port_iter.hasNext()) {
+    auto port = port_iter.next();
+    ports[port->direction()->index()].insert(gui->makeSelected(port));
+  }
+  for (auto dir : {sta::PortDirection::input(),
+                   sta::PortDirection::output(),
+                   sta::PortDirection::tristate(),
+                   sta::PortDirection::bidirect(),
+                   sta::PortDirection::internal(),
+                   sta::PortDirection::ground(),
+                   sta::PortDirection::power(),
+                   sta::PortDirection::unknown()}) {
+    if (!ports[dir->index()].empty()) {
+      std::string direction_str = dir->name();
+      capitalize(direction_str);
+      props.push_back(
+          {fmt::format("{} Ports", direction_str), ports[dir->index()]});
+    }
+  }
+
+  SelectionSet pg_ports;
+  sta::LibertyCellPgPortIterator pg_port_iter(cell);
+  while (pg_port_iter.hasNext()) {
+    pg_ports.insert(gui->makeSelected(pg_port_iter.next()));
+  }
+  props.push_back({"PG Ports", pg_ports});
+
+  return props;
```

Comment:
would also be nice to have the:
* leakagePower
* timing arcs (from -> to)
* cell footprint